celery/canvas.py

```diff
@@ -214,6 +214,7 @@ def apply_async(self, args=None, kwargs=None, route_name=None, **options):
         """
         args = args if args else ()
         kwargs = kwargs if kwargs else {}
+        options = {k: v for k, v in options.items() if v is not None}
```

Comment:
this need comment & unit test as well

Reply:
I'm just wondering if there are use cases out there were you explicitly want to set an option to `None`. That would no longer be supported. I guess onky time will tell.

Reply:
done